### DISCUSSION TITLE
Webauthn user_ids table: Grows indefinitely?

### DISCUSSION THREAD

**Discussion by t1ll** [jeremyevans/rodauth#453] [General]
Hi there,
I noticed that my account_webauthn_user_ids table is ~10x bigger than my account_webauthn_keys table. If I understand it correctly the user_ids table is used during authentication initiation and hence would explain why it's much bigger.

Looking at the docs and the source I was not able to find any method that handles a potential cleanup of that table. Is it correct to assume that it will grow indefinitely?

**Comment by jeremyevans** [comment]
You cannot end up with more than one webauthn user id per account, so it doesn't grow indefinitely, at least no more indefinitely than your accounts table.   If a user closes the account, the related webauthn user id (and any webauthn keys and other authentication data) is deleted.

**Reply by t1ll** [reply]
ahh that makes sense. Thank you for clarifying and sorry for not thinking logically in the first place :)
